crates/turborepo-ci/src/vendors.rs

```diff
@@ -0,0 +1,582 @@
+use std::{collections::HashMap, sync::OnceLock};
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct VendorEnvs {
+    pub(crate) any: Vec<&'static str>,
+    pub(crate) all: Vec<&'static str>,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+#[allow(dead_code)]
+pub struct Vendor {
+    pub(crate) name: &'static str,
+    pub(crate) constant: &'static str,
+    pub(crate) env: VendorEnvs,
+    pub(crate) eval_env: Option<HashMap<&'static str, &'static str>>,
+    pub(crate) sha_env_var: Option<&'static str>,
+    pub(crate) branch_env_var: Option<&'static str>,
+    pub(crate) username_env_var: Option<&'static str>,
+}
+
+static VENDORS: OnceLock<[Vendor; 45]> = OnceLock::new();
+
+pub(crate) fn get_vendors() -> [Vendor; 45] {
```

Comment:
I think we can avoid the clones here if we return static references instead.
```suggestion
pub(crate) fn get_vendors() -> &'static [Vendor] {
```